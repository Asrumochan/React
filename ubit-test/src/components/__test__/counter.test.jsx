import {render,fireEvent,screen} from'@testing-library/react'
import Counter from '../counter'

describe('counter',()=>{
    it("should increase counter",()=>{
        render(<Counter/>)
        const increment=screen.getByTestId("incr")
        const count=screen.getByTestId("count")
        fireEvent.click(increment)
        expect(count).toHaveTextContent("2")
    })
    it("should decrease counter",()=>{
        render(<Counter/>)
        const decrement=screen.getByTestId("decr")
        const count=screen.getByTestId("count")
        fireEvent.click(decrement)
        expect(count).toHaveTextContent("0")
    })
})