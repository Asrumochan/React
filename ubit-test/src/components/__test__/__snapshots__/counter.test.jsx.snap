// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter snapshot should match dom snapshot 1`] = `
<div>
  <button
    data-testid="decr"
    onClick={[Function]}
  >
    -
  </button>
  <p
    data-testid="count"
  >
    Count:
    1
  </p>
  <button
    data-testid="incr"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
